data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -141,26 +144,31 @@ public void initialize() {
     public Optional<SourcePartition<T>> getNextPartition(final Function<Map<String, Object>, List<PartitionIdentifier>> partitionCreationSupplier) {
         validateIsInitialized();
 
-        if (partitionManager.getActivePartition().isPresent()) {
-            return partitionManager.getActivePartition();
-        }
+        Optional<SourcePartitionStoreItem> ownedPartitions = Optional.empty();
+        try {
+          if (lock.tryLock()) {
+            ownedPartitions = sourceCoordinationStore.tryAcquireAvailablePartition(sourceIdentifierWithPartitionType, ownerId, DEFAULT_LEASE_TIMEOUT);
```

Comment:
We should not put a lock around this line. We only want to lock on acquiring the global partition.

So it should look like this

```
Optional<SourcePartitionStoreItem> ownedPartitions = sourceCoordinationStore.tryAcquireAvailablePartition(sourceIdentifierWithPartitionType, ownerId, DEFAULT_LEASE_TIMEOUT);

try {
  if (ownedPartitions.isEmpty && lock.tryLock()) {
      ... code to acquire the global partition and run supplier ... 
  } 
}  finally {
    if (lock.isHeldByCurrentThread()) {
                lock.unlock();
    }

    if (ownedPartition.isEmpty()) {
         ownedPartitions = sourceCoordinationStore.tryAcquireAvailablePartition(sourceIdentifierWithPartitionType, ownerId, DEFAULT_LEASE_TIMEOUT);
    }
}
```